unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -34,17 +36,23 @@ public IEnumerator PopulateRenderersList()
             if (!dirty)
                 yield break;
 
-            renderers = Object.FindObjectsOfType<Renderer>()
-                .Where(x => !(x is SkinnedMeshRenderer)
-                            && !(x is ParticleSystemRenderer)
-                            && ((1 << x.gameObject.layer) & ignoredLayersMask) == 0)
-                .ToArray();
-
-            yield return null;
-
-            skinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>()
-                .Where(x => ((1 << x.gameObject.layer) & ignoredLayersMask) == 0)
-                .ToArray();
+            List<Renderer> renderersList = new List<Renderer>();
+            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
+            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>();
+
+            foreach (Renderer renderer in allRenderers)
+            {
+                if ((((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
+                    && !(renderer is ParticleSystemRenderer))
+                {
+                    if (renderer is SkinnedMeshRenderer)
+                        skinnedRenderersList.Add((SkinnedMeshRenderer)renderer);
+                    else
+                        renderersList.Add(renderer);
+                }
+            }
+            renderers = renderersList.ToArray();
+            skinnedRenderers = skinnedRenderersList.ToArray();
```

Comment:
I think we can improve this even further by reducing the allocations, can we change the type of those fields to `IEnumerator<Renderer>` and `IEnumerator<SkinnedMeshRenderer>`

Reply:
@Kinerius  Of course ^^ I am out of office but tomorrow this changes will be integrated